Define the minimal set of portable functionality (#247)
jakartaee/data: spec/src/main/asciidoc/repository.asciidoc

===== Query Methods Keywords

The following table lists the actions keywords generally supported by Jakarta Data.

|===
|Keyword |Description| Not Required For

|findBy
|General query method returning the repository type.
|Key-value, Wide-Column

|deleteBy
|Delete query method returning either no result (void) or the delete count.
|Key-value, Wide-Column

|countBy
|Count projection returning a numeric result.
|Key-value, Wide-Column

|existsBy
|Exists projection, returning as a `boolean` result.
|Key-value, Wide-Column
|===

[NOTE]
.Note
* The "Not Required For" column indicates the database types for which the respective keyword is not required or applicable.

> Jakarta Data implementations support the following list of predicate keywords to the extent that the database is capable of the behavior. A repository method will raise `java.lang.UnsupportedOperationException` or a more specific subclass of the exception if the database does not provide the requested functionality.

Reviewer: ```suggestion
Jakarta Data implementations must support the following list of query-by-method keywords, except where indicated for a database type. A repository method must raise `java.lang.UnsupportedOperationException` or a more specific subclass of the exception if the database does not provide the requested functionality.
```
